cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +122,142 @@ public Void call() throws Exception {
     }
   }
 
+  /**
+   * Clean log files which does not have corresponding meta data
+   * @param tillTime time till the meta data will be deleted.
+   * @param namespacedLogBaseDirMap namespace to directory map
+   * @param parentDirs parent directories for deleted files
+   * @throws Exception
+   */
+  @VisibleForTesting
+  void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                             final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
```

Comment:
nit: just call it ``namespaces``. The datatype already indicates that its a list.